src/main/java/org/elasticsearch/search/aggregations/AggregatorFactories.java

```diff
@@ -49,8 +44,8 @@ private AggregatorFactories(AggregatorFactory[] factories) {
         this.factories = factories;
     }
 
-    private static Aggregator createAndRegisterContextAware(AggregationContext context, AggregatorFactory factory, Aggregator parent, long estimatedBucketsCount) {
-        final Aggregator aggregator = factory.create(context, parent, estimatedBucketsCount);
+    private static Aggregator createAndRegisterContextAware(AggregationContext context, AggregatorFactory factory, Aggregator parent, boolean collectsSingleBucket) throws IOException {
+        final Aggregator aggregator = factory.create(context, parent, collectsSingleBucket);
```

Comment:
Sorry to comment on the same thing again but now i see it I'm wondering if `collectsFromSingleBucket` would be better since `collectsSingleBucket` seems a bit ambiguous as to whether it collects from a single parent bucket or only collects one bucket in total (which the terms agg doesn't)